internal/wasi/wasi_test.go

```diff
@@ -1874,8 +1874,76 @@ func TestSnapshotPreview1_PollOneoff(t *testing.T) {
 
 func TestSnapshotPreview1_ProcExit(t *testing.T) {
 	ctx := context.Background()
-	sys, err := newSysContext(nil, nil, nil)
-	require.NoError(t, err)
+
+	// Note: Unlike most tests, this uses fn, not the 'a' result parameter. This is because currently, ProcExit function
+	// body panics, and we expect Call to unwrap the panic as an error.
+	// Note: fn are called in various ways to test that it closes the module correctly in various tricky calls.
+	type procExitFn func(exitcode uint32) error
+	procExitFns := []struct {
+		name     string
+		procExit func() (procExitFn, *wasm.Store)
+	}{
+		// call proc_exit in a usual way, that is, call the imported-and-exported-back ProcExit.
+		{FunctionProcExit, func() (procExitFn, *wasm.Store) {
+			sys, err := newSysContext(nil, nil, nil)
+			require.NoError(t, err)
+
+			_, store, mod, fn := instantiateModule(t, context.Background(), FunctionProcExit, ImportProcExit, moduleName, sys)
```

Comment:
AFAICT, the signature change to return a 4th result is only used here. can you revert it and instantiate this test case different way? That prevents drifting almost all tests in this file.